__tests__/OmnichannelChatSDK.spec.ts

```diff
@@ -3265,6 +3265,87 @@ describe('Omnichannel Chat SDK', () => {
             expect(context.reconnectId).toBe(mockedResponse.reconnectid);
         });
 
+
+        it('ChatSDK.getChatReconnectContext() with authenticatedUserToken should call OCClient.getReconnectableChats() & return redirectURL due to expired token', async () => {
+            const chatSDKConfig = {
+                telemetry: {
+                    disable: true
+                },
+                chatReconnect: {
+                    disable: false,
+                }
+            };
+            const params = {
+                reconnectId: 'reconnectId'
+            };
+
+            const chatSDK = new OmnichannelChatSDK(omnichannelConfig, chatSDKConfig);
+            chatSDK.getChatConfig = jest.fn();
+            chatSDK.isChatReconnect = true;
+            chatSDK.authenticatedUserToken = 'token';
+
+            await chatSDK.initialize();
+  
+            const mockedResponse = {
+                reconnectid: 'reconnectid'
+            };
+
+            const mockedResponseAvailability = {
+                reconnectid: null,
+                reconnectRedirectionURL: null
+            };
+
+            jest.spyOn(chatSDK.OCClient, 'getReconnectAvailability').mockResolvedValue(Promise.resolve(mockedResponseAvailability));
+            jest.spyOn(chatSDK.OCClient, 'getReconnectableChats').mockResolvedValue(Promise.resolve(mockedResponse));
+
+            const context = await chatSDK.getChatReconnectContext(params);
+
+            expect(chatSDK.OCClient.getReconnectAvailability).toHaveBeenCalledTimes(1);
+            expect(chatSDK.OCClient.getReconnectableChats).toHaveBeenCalledTimes(1);
```

Comment:
In this situation, if `reconnectRedirectionURL` returns an URL, then getReconnectableChats should not be called

Reply:
![image](https://github.com/microsoft/omnichannel-chat-sdk/assets/981914/9f416d33-7dab-402a-8178-f48dadd24e76)